src/collective/cover/browser/compose.py

```diff
@@ -1,172 +1,59 @@
-
-
-class SetWidgetMap(BrowserView):
-
-    def render(self):
-        widget_map = self.request.get('widget_map')
-        remove = self.request.get('remove', None)
-        self.context.set_widget_map(widget_map, remove)
-        return json.dumps('success')
-
-    def __call__(self):
-        return self.render()
-
-
-class UpdateWidget(BrowserView):
-
-    def render(self):
-        widget_id = self.request.get('wid')
-        if widget_id in self.context:
-            return self.context[widget_id].render()
-        else:
-            return 'Widget does not exist'
-
-    def __call__(self):
-        return self.render()
-
-
-class RemoveTileWidget(BrowserView):
-    # XXX: This should be part of the plone.app.tiles package or similar
-
-    index = ViewPageTemplateFile('templates/removetilewidget.pt')
-
-    def render(self):
-        template = self.template
-        if 'form.submitted' not in self.request:
-            return template.render(self)
-
-        annotations = IAnnotations(self.context)
-        current_tiles = annotations.get('current_tiles', {})
-        tile_id = self.request.get('wid', None)
-
-        if tile_id in current_tiles:
-            widget_type = current_tiles[tile_id]['type']
-            # Let's remove all traces of the value stored in the tile
-            widget_uri = '@@{0}/{1}'.format(widget_type, tile_id)
-            tile = self.context.restrictedTraverse(widget_uri)
-
-            dataManager = ITileDataManager(tile)
-            dataManager.delete()
-        return self.index()
-
-    def __call__(self):
-        return self.render()
-
 
 class UpdateTileContent(BrowserView):
 
     """Helper browser view to update the content of a tile."""
 
+    def setup(self):
+        self.tile_type = self.request.form.get('tile-type')
+        self.tile_id = self.request.form.get('tile-id')
+        self.uuid = self.request.form.get('uuid')
+
     def render(self):
         """Render a tile after populating it with an object."""
-        tile_type = self.request.form.get('tile-type')
-        tile_id = self.request.form.get('tile-id')
-        uuid = self.request.form.get('uuid')
-        if tile_type and tile_id and uuid:
+        if self.tile_type and self.tile_id and self.uuid:
             catalog = api.portal.get_tool('portal_catalog')
-            results = catalog(UID=uuid)
+            results = catalog(UID=self.uuid)
             if results:
                 obj = results[0].getObject()
-                tile = self.context.restrictedTraverse('{0}/{1}'.format(tile_type, tile_id))
+                tile = self.context.restrictedTraverse('{0}/{1}'.format(self.tile_type, self.tile_id))
                 tile.populate_with_object(obj)
                 # reinstantiate the tile to update its content on AJAX calls
-                tile = self.context.restrictedTraverse('{0}/{1}'.format(tile_type, tile_id))
+                tile = self.context.restrictedTraverse('{0}/{1}'.format(self.tile_type, self.tile_id))
```

Comment:
@rodfersou wouldn't be better to do `'{0}/{1}'.format(self.tile_type, self.tile_id)` only once?